silk-core/src/main/kotlin/net/silkmc/silk/core/serialization/serializers/VectorSerializer.kt

```diff
@@ -0,0 +1,106 @@
+package net.silkmc.silk.core.serialization.serializers
+
+import com.mojang.math.Vector3d
+import com.mojang.math.Vector3f
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerializationStrategy
+import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.encoding.CompositeDecoder
+import kotlinx.serialization.encoding.CompositeEncoder
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import net.minecraft.core.Vec3i
+import net.silkmc.silk.core.math.vector.triple
+import net.silkmc.silk.core.serialization.SilkSerializer
+
+private const val XIndex = 0;
+private const val YIndex = 1;
+private const val ZIndex = 2;
```

Comment:
```suggestion
private const val xIndex = 0;
private const val yIndex = 1;
private const val zIndex = 2;
```
and don't forget to update the usages as well